gltfTutorial/gltfTutorial_006_MaterialsTechniquesProgramsShaders.md

```diff
@@ -0,0 +1,288 @@
+
+There are several well-known parameters that are commonly used to describe a material in OpenGL:
+
+* The "diffuse" color: The main color of the material, which is often defined by a texture
+* The "specular" color: The color of specular highlights
+* The "shininess" factor: A value that defines the shininess of the object, influencing the size of the specular highlights
+* The "emissive" color: The color of the light that is emitted by the object
+* ...
+
+Many file formats contain this information in one or the other form. For example, [Wavefront OBJ](https://en.wikipedia.org/wiki/Wavefront_.obj_file) files are combined with `MTL` files that contain exactly these parameters. Viewers can read this information and render the objects accordingly.
+
+However, one of the goals of glTF was to **not** constrain the rendering to one simple, fixed material model. It should be possible to encapsulate arbitrary rendered scenes in one glTF asset, *including* the information about how exactly the objects are rendered. This is an ambitious goal: There are unlimited degrees of freedom when rendering with OpenGL. To retain this flexibility, the glTF specification of materials can be considered as a very generic description of rendering processes. On the lowest level, it encapsulates the actual GLSL shader programs, and thus can cover nearly all imaginable rendering pipelines.
+
+This flexibility comes at a certain cost. Several elements in the glTF asset have to be combined properly by the renderer to exploit this flexibility. The following sections will break down these elements and describe how they are combined and interpreted to finally render an asset with OpenGL. The description will be given in a bottom-up fashion: It will start with a short summary of how GLSL shaders are used for rendering in OpenGL, and show how these shaders are encoded in glTF. Then, it will show how these shader programs serve as the basis for describing a generic rendering technique. Finally, it will give describe the process of rendering an object based on a the material definition that was given in the glTF.
+
+
+## Programs and Shaders
+
+The rendering process in OpenGL is described with small shader programs, which are written in [GLSL, the OpenGL shading language](https://www.opengl.org/documentation/glsl/). Such a shader program consists of at least two shaders, namely the vertex shader and the fragment shader. Roughly speaking, the vertex shader defines how the vertices of an object are transformed, preprocessed and projected on the screen, and the fragment shader defines how these projected vertices are used to fill the pixels on the screen to generate the final rendered image. For further details, refer to one of the GLSL tutorials (e.g. at [open.gl](https://open.gl/drawing#Shaders), [webglfundamentals.org](http://webglfundamentals.org/webgl/lessons/webgl-shaders-and-glsl.html) or others)

```

Comment:
The fragment shader determines the color (and potentially depth) of fragments.  Fragments become pixels when they pass the depth test by the end of the frame.

The explanation above kind of sounds like rasterizations happens as part of the fragment shader.